src/test/java/org/folio/rest/impl/FileUploadServiceTest.java

```diff
@@ -62,6 +62,24 @@ public void postFileDefinition_return200Status(TestContext context) {
     async.complete();
   }
 
+  @Test
+  public void postFileDefinition_return422Status_whenFileNameExtensionNotCsv(TestContext context) {
+    Async async = context.async();
+    // given file
+    FileDefinition givenFileDefinition = new FileDefinition()
+      .withId(UUID.randomUUID().toString())
+      .withFileName("InventoryUUIDs.txt");
+    // when created a new entity
+    Response response = RestAssured.given()
+      .spec(jsonRequestSpecification)
+      .body(JsonObject.mapFrom(givenFileDefinition).encode())
+      .when()
+      .post(FILE_DEFINITION_SERVICE_URL);
+    // then retrieve it and verify
+    context.assertEquals(HttpStatus.SC_UNPROCESSABLE_ENTITY, response.getStatusCode());
```

Comment:
Also verify the message returned is the same one that is being thrown, so we know it doesn't fail with a different error

Reply:
done